tracing/providers/opentelemetry/src/main/java/io/helidon/tracing/providers/opentelemetry/OpenTelemetrySpan.java

```diff
@@ -90,19 +95,33 @@ public void addEvent(String name, Map<String, ?> attributes) {
     @Override
     public void end() {
         delegate.end();
+        spanLifeCycleListeners.forEach(listener -> listener.ended(limited()));
     }
 
     @Override
     public void end(Throwable t) {
         delegate.recordException(t);
         delegate.setStatus(StatusCode.ERROR);
         delegate.end();
+        spanLifeCycleListeners.forEach(listener -> listener.ended(limited(), t));
     }
 
     @Override
     public Scope activate() {
         io.opentelemetry.context.Scope scope = otelContextWithSpanAndBaggage().makeCurrent();
-        return new OpenTelemetryScope(scope);
+        var result = new OpenTelemetryScope(this, scope, spanLifeCycleListeners);
+        UnsupportedActivationException ex = new UnsupportedActivationException("Error activating span", result);
```

Comment:
It would be nice to avoid instantiating and closing over an `UnsupportedActivationException` in all cases, even successful ones. I understand what you're doing with suppressions etc. but exception creation is pretty heavyweight.

Reply:
Fixed.